### PR TITLE
Adding data-stream related permissions

### PR DESCRIPTION
**Pull request by sandeshkr419**
Signed-off-by: Sandesh Kumar <sandeshkr419@gmail.com>

##  opensearch-project/security-dashboards-plugin pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Enhancement
 
 
 2. __Github Issue # or road-map entry, if available:__
https://github.com/opensearch-project/security-dashboards-plugin/issues/782

 3. __Description of changes:__
Update list of permissions displayed in UI with data-stream related permissions.

 4. __Why these changes are required?__ 
So that the data-stream related permissions are displayed in UI Permissions list.

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)
 No permissions were displayed in UI for data-stream related tasks. User had to manually type in the desired permissions.

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
Testing of the permission groups added in this change was already done as part of https://github.com/opensearch-project/security/pull/1170
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)
No

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/apps/configuration/constants.tsx [hunk 1]
```diff
@@ -138,12 +138,18 @@ export const INDEX_PERMISSIONS: string[] = [
...
+  'indices:admin/data_stream/create',
+  'indices:admin/data_stream/delete',
+  'indices:admin/data_stream/get',
   'indices:admin/delete',
   'indices:admin/exists',
   'indices:admin/flush',
   'indices:admin/flush*',
   'indices:admin/forcemerge',
   'indices:admin/get',
+  'indices:admin/index_template/delete',
```
Comment (vrozov) [inline public/apps/configuration/constants.tsx:150]: any other index_template actions?

Comment (sandeshkr419) [reply public/apps/configuration/constants.tsx:150]: I have checked thoroughly Opensearch code to get exhaustive list of operations supported w.r.t. index templates. I have found only 3 actions which I have included them here.

Comment (vrozov) [reply public/apps/configuration/constants.tsx:150]: `indices:admin/index_template/simulate` and `indices:admin/index_template/simulate_index`?

Comment (sandeshkr419) [reply public/apps/configuration/constants.tsx:150]: Seems some issue with my editor, rechecked again. These 2 actions specified by you complete the list of all supported actions for index templates. Adding them in this list.

**Review by vrozov — APPROVED** [review]

**Review by vrozov — APPROVED** [review]

**Review by cliu123 — APPROVED** [review]

**Pull request merged**